core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -91,7 +90,12 @@ public static void dropTableData(FileIO io, TableMetadata metadata) {
     Set<ManifestFile> manifestsToDelete = Sets.newHashSet();
     for (Snapshot snapshot : metadata.snapshots()) {
       // add all manifests to the delete set because both data and delete files should be removed
-      Iterables.addAll(manifestsToDelete, snapshot.allManifests(io));
+      try {
+        Iterables.addAll(manifestsToDelete, snapshot.allManifests(io));
+      } catch (RuntimeException e) {
```

Comment:
Should we make this one less wide? This might catch a lot of exceptions and hide certain issues. How about using the KeyNotFoundException as mentioned in the issue?

Reply:
`NoSuchKeyException` is very much storage specific. We will have come with a approach to generalize it for all the file systems like **(S3/Azure/GCS)** . 
Please share your thoughts.